plugins/webp-uploads/helper.php

```diff
@@ -144,7 +145,7 @@ function webp_uploads_generate_additional_image_source( int $attachment_id, stri
 		return new WP_Error( 'image_mime_type_not_supported', __( 'The provided mime type is not supported.', 'webp-uploads' ) );
 	}
 
-	$image_path = wp_get_original_image_path( $attachment_id );
+	$image_path = 'application/pdf' !== get_post_mime_type( $attachment_id ) ? wp_get_original_image_path( $attachment_id ) : get_attached_file( $attachment_id );
```

Comment:
Minor point: This is easier to read and it would facilitate ensuring test coverage is present for both code paths. Also, in the future maybe there will be more such special cases.

```suggestion
	if ( 'application/pdf' === get_post_mime_type( $attachment_id ) {
		$image_path = get_attached_file( $attachment_id );	
	} else {
		$image_path = wp_get_original_image_path( $attachment_id );	}
```

Reply:
I was thinking something like this: 
```
$image_path = wp_attachment_is_image( $attachment_id ) ? wp_get_original_image_path( $attachment_id ) : get_attached_file( $attachment_id );
``` 

This way we can use `wp_get_original_image_path()` only when attatchment is an image and in all other cases we can use `get_attached_file ()`

This is also more in accordance with latest suggestions made on the issue thread: https://github.com/WordPress/performance/issues/1766#issuecomment-2653759951